core/foundation/test/testStringUtils.cxx

```diff
@@ -20,4 +20,23 @@ TEST(StringUtils, Split)
    test(",,,", {}, true);
    test(",,a", {"", "", "a"}, false);
    test("", {""}, false);
+
+   // Test that ROOT::Join behaves like str.join from Python.
+
+   /*
+   auto test_join_cxx20 = []<typename T>(const std::string res, const std::string& sep, T&&...strings) {
+      auto out = ROOT::Join(sep, strings);
+      EXPECT_EQ(out, ref) << "ROOT::Join(\"" << sep << "\") gave wrong result.";
+   };
+   */
+   auto test_join = [](const std::string ref, const std::string& sep, std::vector<std::string> const &strings) {
+      auto out = ROOT::Join(sep, strings);
+      EXPECT_EQ(out, ref) << "ROOT::Join(\"" << sep << "\") gave wrong result.";
```

Comment:
I don't understand the error message `ROOT::Join(\"" << sep << "\")`, are we sure writing just the `sep` makes the error more clear?